linkerd/app/integration/src/tests/telemetry/tcp_errors.rs

```diff
@@ -235,6 +276,42 @@ async fn inbound_direct_multi() {
     drop(tcp_client);
 }
 
+#[tokio::test]
+async fn inbound_invalid_ip() {
+    let _trace = trace_init();
+    let srv = server::http2().route("/", "hello world").run().await;
+    let fake_ip = SocketAddr::from(([127, 0, 0, 1], 9999));
+    let (proxy, metrics) = Test::new(proxy::new().inbound(srv).inbound_ip(fake_ip))
+        .with_extra_env(|env| {
+            env.put(
+                app::env::ENV_INBOUND_IPS,
+                "10.5.1.42,10.5.8.167".to_string(),
+            );
+        })
+        .run()
+        .await;
+
+    let client = client::tcp(proxy.inbound);
+    let metric = metric(&proxy)
+        .label("error", "invalid ip address")
+        .label("target_addr", fake_ip);
+
+    let tcp_client = client.connect().await;
+    tcp_client.write(TcpFixture::HELLO_MSG).await;
+    drop(tcp_client);
+
+    metric.clone().value(1u64).assert_in(&metrics).await;
+
+    // this connectiopn *would* time out, but it doesn't hit the timeout,
+    // because the invalid target addr will already have been rejected.
+    let tcp_client = client.connect().await;
+    tokio::time::sleep(TIMEOUT + Duration::from_millis(15)) // just in case
+        .await;
+    drop(tcp_client);
+
+    metric.clone().value(2u64).assert_in(&metrics).await;
+}
```

Comment:
rather than use metrics to test this, it's probably preferable to test the inbound accept stack -- that it actually fails with an error.

Reply:
This was specifically intended as a test _for_ the error metrics being updated, not for the behavior overall. We should probably add stack tests separately.

If you don't think it's worthwhile to test that we emit the correct metric, this can also be removed?

Reply:
we could check that the unexpected error is incremented here... don't feel strongly.